src/trackinfoobject.cpp

```diff
@@ -722,20 +722,28 @@ QString TrackInfoObject::getURL() {
 }
 
 ConstWaveformPointer TrackInfoObject::getWaveform() {
+    QMutexLocker lock(&m_qMutex);
```

Comment:
Locking is not necessary here. Accessing the native pointer wrapped in QSharedPointer is thread safe. See also: Qt Docs > QSharedPointer > Thread-Safety